### ISSUE TITLE
Use tf.data and tf.keras in multitasking (each task uses a different custom loss)

### ISSUE THREAD

**Issue by liupengkd** [tensorflow/tensorflow#39224]
**System information**
- OS Platform and Distribution : Linux Ubuntu 16.04
- TensorFlow installed from (source or binary):binary
- TensorFlow version (use command below):2.0
- Python version:3.6

**description**
**step 1)** I used tf.keras to build a single-input dual-output multi-task model, each output loss is customized (by inheriting tf.keras.losses.Loss).
<img width="174" alt="image" src="https://user-images.githubusercontent.com/15835439/81172546-069b4f80-8fd1-11ea-881a-93fecf39992f.png">
**step 2)** Construct a data set through tf.data-(image, (label1, label2)).
**step 3)** 
<img width="359" alt="image" src="https://user-images.githubusercontent.com/15835439/81172981-d7d1a900-8fd1-11ea-97c9-85933f1deaa4.png">
<img width="176" alt="image" src="https://user-images.githubusercontent.com/15835439/81172995-df914d80-8fd1-11ea-9852-aff980257594.png">
**bug**

I found that the two outputs of the network were sent to the corresponding loss functions, but the corresponding label1 and label2 in tf.data were not sent to the two loss functions

**Comment by Saduf2019** [comment]
@liupengkd 
Can you please share simple stand alone code for us to replicate the issue or is possible share a colab gist for us to analyse the issue faced.

**Comment by liupengkd** [comment]
@Saduf2019 
code like this：
···python
 def test(input_shape=(None,None,3), is_training=True):                                                                                                      
     inputs = tf.keras.layers.Input(shape=input_shape)#one input                                                                                             
     '''                                                                                                                                                     
     Various network layers                                                                                                                                  
     '''                                                                                                                                                     
     #two outputs                                                                                                                                            
     predict1 = ...                                                                                                                                          
     predict2 = ...                                                                                                                                          
                                                                                                                                                             
     model = tf.keras.models.Model(inputs=inputs, outputs=[predict1, preidct2])                                                                              
     model.summary()                                                                                                                                         
                                                                                                                                                             
     return model                                                                                                                                            
                                                                                                                                                             
 datas = [...]                                                                                                                                               
 def process(l):                                                                                                                                             
     '''                                                                                                                                                     
     Process each piece of data to get training data, label1 and label2                                                                                      
     '''                                                                                                                                                     
                                                                                                                                                             
     return data, (label1, label2)                                                                                                                           
                                                                                                                                                             
 class loss1(tf.keras.losses.Loss):
     def __init__(self):                                                                                                                                     
         super().__init__()                                                                                                                                  
                                                                                                                                                             
     def call(self, label1, predict1):                                                                                                                       
         '''                                                                                                                                                 
         Processing to get loss1                                                                                                                             
         '''                                                                                                                                                 
                                                                                                                                                             
         return loss1                                                                                                                                        
                                                                                                                                                             
 class loss2(tf.keras.losses.Loss):                                                                                                                          
                                                                                                                                                             
     def __init__(self):                                                                                                                                     
         super().__init__()                                                                                                                                  
                                                                                                                                                             
     def call(self, label2, predicts2):                                                                                                                      
         '''                                                                                                                                                 
         Processing to get loss2                                                                                                                             
         '''                                                                                                                                                 
                                                                                                                                                             
         return loss2

                                                                                                                                                             
 train_ds = tf.data.Dataset.from_tensor_slices(datas).map(process, 8)                                                                                        
                                                                                                                                                             
 model.compile(                                                                                                                                              
     optimizer=tf.keras.optimizers.SGD(learning_rate=0.001, momentum=0.9, nesterov=True),                                                                    
     loss=[loss1(), loss2()],                                                                                                                                
     loss_weights=[0.5, 1.])                                                                                                                                 
                                                                                                                                                             
                                                                                                                                                             
 model.fit(train_ds,epochs=100) 
···

I found that the two outputs of the network（predict1 and predict2） were sent to the corresponding loss functions, but the corresponding label1 and label2 in tf.data were not sent to the two loss functions

**Comment by Saduf2019** [comment]
@liupengkd 
Could you please provide with indented stand alone code such that we could replicate the issue faced, i ran the code shared and [here is the gist](https://colab.sandbox.google.com/gist/Saduf2019/a5b0d3d40d416dd5dd03ce43a7a7182f/untitled166.ipynb)

**Comment by Saduf2019** [comment]
@liupengkd
Please update as per above comment

**Comment by liupengkd** [comment]
@Saduf2019 I have changed another training method，not tf.keras.
